### DISCUSSION TITLE
Deprecation of Feeds API support for XML and Flat File Listings Feeds

### DISCUSSION THREAD

**Discussion by waldrogi** [amzn/selling-partner-api-models#3815] [Announcements]
Starting today, we are deprecating XML and flat file listings data (including pricing, inventory, relationships and images) submitted via the Feeds API and will no longer provide new features or categories in the legacy category XML format.  If you have an application that uses XML and flat file listings data, you have until March 31, 2025 to migrate to the JSON-based Listings Items APIs or the `JSON_LISTINGS_FEED` submitted via the Feeds API. This deprecation does not impact any listings feeds submitted by sellers directly through Seller Central.

In 2021, we launched JSON-based Listings APIs and the `JSON_LISTINGS_FEED` feed type for the Feeds API. These solutions empower developers to access new listings capabilities and are available for all product types in hardlines, softlines, and consumables categories worldwide. With these solutions, developers can build sophisticated workflows to create and manage listings. This change applies to all stores worldwide.

The key benefits include: 

- Up-to-date and complete data requirements in product schemas, accessible via the [Product Type Definitions](https://developer-docs.amazon.com/sp-api/docs/product-type-api-use-case-guide) API.
- Uniformity in schemas across data requirement.
- Complete view of listing errors, accessible synchronously and asynchronously.
- Validation-mode submissions through [Listings Items Put/Patch operations](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-use-case-guide) to validate listing data without submitting it to the Amazon catalog.
- Push notifications for listing status changes and listing issues.

**Feed types that will be sunset on March 31, 2025:**

- `POST_PRODUCT_DATA` 
- `POST_INVENTORY_AVAILABILITY_DATA` 
- `POST_PRODUCT_OVERRIDES_DATA`
- `POST_PRODUCT_PRICING_DATA`
- `POST_PRODUCT_IMAGE_DATA`
- `POST_PRODUCT_RELATIONSHIP_DATA` 
- `POST_FLAT_FILE_INVLOADER_DATA`
- `POST_FLAT_FILE_LISTINGS_DATA`
- `POST_FLAT_FILE_BOOKLOADER_DATA`
- `POST_FLAT_FILE_CONVERGENCE_LISTINGS_DATA`
- `POST_FLAT_FILE_LISTINGS_DATA`
- `POST_FLAT_FILE_PRICEANDQUANTITYONLY_UPDATE_DATA`
- `POST_UIEE_BOOKLOADER_DATA`

## What action is required?

If you have an application that uses any of the feed types listed above, you have until March 31, 2025 to migrate to the JSON-based Listings Items APIs or the `JSON_LISTINGS_FEED` submitted via the Feeds API.

You may notice an increase in product attribute requirements in JSON Schemas from the Product Type Definitions API when compared to category XSDs. These changes more accurately represent the actual requirements for Selling Partners (including when attributes are required conditionally based on other listing data provided). Developers are empowered to accurately meet listing requirements before submitting to Amazon, reducing errors previously received after submission.

Feeds API usage for the feed types listed above will enter a grace period early next year, at which time throughput may be degraded.

## Planned new features

Support for Amazon Business (B2B) pricing, media categories, and digital categories coming later this year. In addition to the current features of the JSON-based Listings APIs and the `JSON_LISTINGS_FEED` feed type, we will soon launch new features that further empower developers to enhance their listings workflows. 

## For more information

The guides, videos, and sample application below have been developed to help you migrate your applications. 

Workflow Guide and YouTube videos on Listings Items API:

- [Building Listings Management Workflows Guide](https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide)
- [Listings Items API Use Case Guide](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-use-case-guide)
- [Listings Items API issue troubleshooting](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-issues-troubleshooting)
- [Listings API YouTube video series](https://www.youtube.com/watch?v=-box2_OmQpE)

Migration from category XSD/flat files to JSON

- [Migrating Listings Management Workflows Guide](https://developer-docs.amazon.com/sp-api/docs/migrating-listings-management-workflows)
- [Listings XSD to JSON Migration Webinar](https://www.youtube.com/watch?v=MJHQPXWwGWo)
- [Metadata schema mapping files](https://developer-docs.amazon.com/sp-api/docs/migrating-listings-management-workflows#mapping-product-attributes-from-category-xsds-xml-and-flat-file-feed-types)

Sample Solution

- [Listings Items API Sample solution GitHub link](https://github.com/amzn/selling-partner-api-samples/tree/main/use-cases/listings)
- [Listings Sample Solution Setup & Installation](https://www.youtube.com/watch?v=dGmlveknTks)
- [Listings Sample Solution Create Listing Flow](https://www.youtube.com/watch?v=65nN8JmIc9s)

If you need help, please reach out to our [Developer Support team](https://developer.amazonservices.com/support).

**Comment by limeng92** [comment]
Hello, I am using your API document to upload JSON format data with feed type JSON_LISTINGS_FEED to batch update the inventory data of sku. The document address is: https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-use-case-guide. However, when I use the relevant API to batch update sku, I encounter related errors. Can you help me check why the update fails? The following are the relevant request body and response body, as well as the error message.[
[request_info.txt](https://github.com/user-attachments/files/16942392/request_info.txt)
](url)

**Reply by mafge** [reply]
Hi @limeng92,
thanks for reaching out!
Your feed looks okay to me, can you try to use a more specific product type and not the generic `PRODUCT` one?
Best,
Marc
Selling Partner Developer Services

**Reply by limeng92** [reply]
thanks

**Reply by lmarkenson** [reply]
> Hi @limeng92, thanks for reaching out! Your feed looks okay to me, can you try to use a more specific product type and not the generic `PRODUCT` one? Best, Marc Selling Partner Developer Services

Hello Amazon team,

Why can't we use the generic product type "PRODUCT" alongside PARTIAL_UPDATE to update quantity and price? If perform a PATCH with a REPLACE OP, would the generic product type "PRODUCT" be a problem when trying to submit quantity and price update? 

Thank you!

**Comment by limeng92** [comment]
Hello, I update the inventory by calling your official document feed (type: JSON_LISTINGS_FEED) API to submit SKUs in batches, but it prompts that the quota is insufficient. There are 15 tasks, and half of them failed due to quota. How can I solve this problem? Can the quota be increased?
[request_info.txt](https://github.com/user-attachments/files/17000551/request_info.txt)

**Reply by mafge** [reply]
Hi @limeng92,
thanks for reaching out!
Please note that `JSON_LISTINGS_FEED` allows one feed submission every five minutes with up to 10.000 items per feed. There is no way to increase the quota.
Best,
Marc
Selling Partner Developer Services

**Reply by limeng92** [reply]
> Hi @limeng92, thanks for reaching out! Please note that `JSON_LISTINGS_FEED` allows one feed submission every five minutes with up to 10.000 items per feed. There is no way to increase the quota. Best, Marc Selling Partner Developer Services

Hello, I have nearly 15 - 20 tasks to deal with here. If I set it to submit the feed every 5 minutes, it will probably take about 80 minutes, which is a bit too long. Is there a good way to solve this problem? Also, even though I have set the timer to trigger every 5 minutes, there are still one or two tasks occasionally showing quota timeout. What's the problem here? Because our requirement is to scan the tasks and submit the feed at 9:00 and 15:00 every day to update the inventory data. I would like to ask if there is a better way to solve the problem? Preferably, a way that can shorten the time of the tasks. Thank you.

**Reply by mafge** [reply]
@limeng92 There is currently no way to increase the quota. The general recommendation is to make sure to only update the data that really needs to be updated, avoid duplicate submissions, and stick to our [best practices](https://developer-docs.amazon.com/sp-api/docs/feeds-api-best-practices). You can also use Listings Items API in parallel to increase throughput on your end.

**Reply by limeng92** [reply]
> @limeng92 There is currently no way to increase the quota. The general recommendation is to make sure to only update the data that really needs to be updated, avoid duplicate submissions, and stick to our [best practices](https://developer-docs.amazon.com/sp-api/docs/feeds-api-best-practices). You can also use Listings Items API in parallel to increase throughput on your end.

Thank you

**Comment by limeng92** [comment]
Hello, I have nearly 15 - 20 tasks to deal with here. If I set it to submit the feed every 5 minutes, it will probably take about 80 minutes, which is a bit too long. Is there a good way to solve this problem? Also, even though I have set the timer to trigger every 5 minutes, there are still one or two tasks occasionally showing quota timeout. What's the problem here? Because our requirement is to scan the tasks and submit the feed at 9:00 and 15:00 every day to update the inventory data. I would like to ask if there is a better way to solve the problem? Preferably, a way that can shorten the time of the tasks. Thank you.













At 2024-09-16 16:59:47, "mafge" ***@***.***> wrote:

Hi @limeng92,
thanks for reaching out!
Please note that allows one feed submission every five minutes with up to 10.000 items per feed. There is no way to increase the quota.
Best,
Marc
Selling Partner Developer ServicesJSON_LISTINGS_FEED

**Reply by kevnhowe** [reply]
Is this for a single seller and marketplace? Is there no way to combine the update inventory tasks into a single task so that you only make one call each time you want to update inventory?

**Reply by mafge** [reply]
@kevnhowe Without further details on your use case and how you operate this is hard to answer because there are too many variables to consider. Let me give you some general guidance instead:
- For inventory/price update I would not encourage you to use Feeds API, because in most cases you want to get the change applied as soon as possible. Using patch operation from Listings Items API is the better way for these cases. For creating listings, general updates that you run infrequently and are not time critical, you can use Feeds API and submit these changes in bulk.

**Reply by kevnhowe** [reply]
> @kevnhowe Without further details on your use case and how you operate this is hard to answer because there are too many variables to consider. Let me give you some general guidance instead:
> 
> * For inventory/price update I would not encourage you to use Feeds API, because in most cases you want to get the change applied as soon as possible. Using patch operation from Listings Items API is the better way for these cases. For creating listings, general updates that you run infrequently and are not time critical, you can use Feeds API and submit these changes in bulk.

I was replying to @limeng92 's question. It is their use case. I was suggesting that, if possible, their tasks can be combined to send a single inventory update request, instead of iterating over their tasks and waiting for the throttle limiter for each call.

**Comment by limeng92** [comment]
Thank you for your reply. I want to ask, if there are multiple stores and multiple markets, how should I completely update the entire inventory without using the feed API? The Listings Items API seems to be able to update the inventory of only a single sku.













At 2024-10-19 05:46:29, "kevnhowe" ***@***.***> wrote:

Is this for a single seller and marketplace? Is there no way to combine the update inventory tasks into a single task so that you only make one call each time you want to update inventory?

**Reply by mafge** [reply]
Hi @limeng92,
yes, Listings Items API is a one by one operation, which means you can only update one SKU with each call. This means you need to  make multiple calls to update multiple SKUs. However, if you are dealing with a high number of updates throughout the day, my recommendation is to think about a more dynamic way of updating your listings. Meaning moving away from a strict schedule of updating prices/inventory once a day at a specific time to a more flexible schedule and doing updates only if really necessary. For price updates you can e.g. subscribe to our pricing notifications and only update prices if really necessary. This will help you to avoid running into rate limits and it also helps us to speed up the processing of your updates in the backend.

**Comment by limeng92** [comment]
OK, thanks for your reply. Updating a single sku may not be appropriate, because we have thousands of sku in one store and several markets, and it is stipulated that all quantities must be updated in four time periods every day. At present, I have found that it can only be updated through the feed api, which seems to be the only way.










At 2024-10-25 15:02:59, "mafge" ***@***.***> wrote:

Hi @limeng92,
yes, Listings Items API is a one by one operation, which means you can only update one SKU with each call. This means you need to make multiple calls to update multiple SKUs. However, if you are dealing with a high number of updates throughout the day, my recommendation is to think about a more dynamic way of updating your listings. Meaning moving away from a strict schedule of updating prices/inventory once a day at a specific time to a more flexible schedule and doing updates only if really necessary. For price updates you can e.g. subscribe to our pricing notifications and only update prices if really necessary. This will help you to avoid running into rate limits and it also helps us to speed up the processing of your updates in the backend.
